alpine/FreeElectronLaser.cpp

```diff
@@ -0,0 +1,1851 @@
+  {0.96904,0.43399,0.10294},
+  {0.96555,0.42241,0.09798},
+  {0.96187,0.41093,0.09310},
+  {0.95801,0.39958,0.08831},
+  {0.95398,0.38836,0.08362},
+  {0.94977,0.37729,0.07905},
+  {0.94538,0.36638,0.07461},
+  {0.94084,0.35566,0.07031},
+  {0.93612,0.34513,0.06616},
+  {0.93125,0.33482,0.06218},
+  {0.92623,0.32473,0.05837},
+  {0.92105,0.31489,0.05475},
+  {0.91572,0.30530,0.05134},
+  {0.91024,0.29599,0.04814},
+  {0.90463,0.28696,0.04516},
+  {0.89888,0.27824,0.04243},
+  {0.89298,0.26981,0.03993},
+  {0.88691,0.26152,0.03753},
+  {0.88066,0.25334,0.03521},
+  {0.87422,0.24526,0.03297},
+  {0.86760,0.23730,0.03082},
+  {0.86079,0.22945,0.02875},
+  {0.85380,0.22170,0.02677},
+  {0.84662,0.21407,0.02487},
+  {0.83926,0.20654,0.02305},
+  {0.83172,0.19912,0.02131},
+  {0.82399,0.19182,0.01966},
+  {0.81608,0.18462,0.01809},
+  {0.80799,0.17753,0.01660},
+  {0.79971,0.17055,0.01520},
+  {0.79125,0.16368,0.01387},
+  {0.78260,0.15693,0.01264},
+  {0.77377,0.15028,0.01148},
+  {0.76476,0.14374,0.01041},
+  {0.75556,0.13731,0.00942},
+  {0.74617,0.13098,0.00851},
+  {0.73661,0.12477,0.00769},
+  {0.72686,0.11867,0.00695},
+  {0.71692,0.11268,0.00629},
+  {0.70680,0.10680,0.00571},
+  {0.69650,0.10102,0.00522},
+  {0.68602,0.09536,0.00481},
+  {0.67535,0.08980,0.00449},
+  {0.66449,0.08436,0.00424},
+  {0.65345,0.07902,0.00408},
+  {0.64223,0.07380,0.00401},
+  {0.63082,0.06868,0.00401},
+  {0.61923,0.06367,0.00410},
+  {0.60746,0.05878,0.00427},
+  {0.59550,0.05399,0.00453},
+  {0.58336,0.04931,0.00486},
+  {0.57103,0.04474,0.00529},
+  {0.55852,0.04028,0.00579},
+  {0.54583,0.03593,0.00638},
+  {0.53295,0.03169,0.00705},
+  {0.51989,0.02756,0.00780},
+  {0.50664,0.02354,0.00863},
+  {0.49321,0.01963,0.00955},
+  {0.47960,0.01583,0.01055}
+};
+
+
+
+
+uint64_t nanoTime(){
+    using namespace std;
+    using namespace chrono;
+    return duration_cast<nanoseconds>(high_resolution_clock::now().time_since_epoch()).count();
+}
+//CONFIG
+KOKKOS_INLINE_FUNCTION bool isNaN(float x){
+    #ifdef __CUDA_ARCH__
+    return isnan(x);
+    #else
+    return std::isnan(x);
+    #endif
+}
+KOKKOS_INLINE_FUNCTION bool isINF(float x){
+    #ifdef __CUDA_ARCH__
+    return isinf(x);
+    #else
+    return std::isinf(x);
+    #endif
+}
+KOKKOS_INLINE_FUNCTION bool isNaN(double x){
+    #ifdef __CUDA_ARCH__
+    return isnan(x);
+    #else
+    return std::isnan(x);
+    #endif
+}
+KOKKOS_INLINE_FUNCTION bool isINF(double x){
+    #ifdef __CUDA_ARCH__
+    return isinf(x);
+    #else
+    return std::isinf(x);
+    #endif
+}
+template<unsigned int Dim, typename callable, typename... Ts>
+KOKKOS_INLINE_FUNCTION void serial_for(callable c, ippl::Vector<uint32_t, Dim> from, ippl::Vector<uint32_t, Dim> to, Ts... args){
+    if constexpr(sizeof...(Ts) == Dim){
+        c(args...);
+    }
+    else{
+        for(uint32_t i = from[sizeof...(Ts)];i < to[sizeof...(Ts)];i++){
+            serial_for(c, from, to, args..., i);
+        }
+    }
+}
```

Comment:
Would it make sense to put this in a helper file since it is not specific to the application itself?